### PR TITLE
Make System.Drawing code compatible with .NET Core

### PR DESCRIPTION
**Pull request by qmfrederik**
Some low hanging fruit which makes the System.Drawing code "more" compatible with .NET Core.

### CODE REVIEW

File: mcs/class/System.Drawing/System.Drawing/Bitmap.cs [hunk 1]
```diff
@@ -133,7 +134,7 @@ public Bitmap (Type type, string resource)
 			if (resource == null)
 				throw new ArgumentException ("resource");
 
-			Stream s = type.Assembly.GetManifestResourceStream (type, resource);
+			Stream s = type.GetTypeInfo ().Assembly.GetManifestResourceStream (type, resource);
```
Comment (marek-safar) [inline mcs/class/System.Drawing/System.Drawing/Bitmap.cs]: This breaks existing tests

**Comment by qmfrederik** [comment]
@marek-safar Very odd GetTypeInfo() doesn't work. I've reverted that part of this PR for now.

**Comment by akoeplinger** [comment]
@qmfrederik it does work, it's just that we have tests that verify you get NullReferenceException when passing null as the first argument to e.g. the `Bitmap(Type, string)` constructor (which is terrible, but it's what .NET does, most likely due to an implementation bug).

But the added .GetTypeInfo() has a check that checks for null and throws an ArgumentNullException instead so that test fails.

So we can do two things: 1) deviate from .NET and change the test to expect ArgumentNullException, 2) explicitly check for null and throw a NullReferenceException.

**Comment by marek-safar** [comment]
Fixing the test to expect ArgumentNullException is ok here

**Comment by qmfrederik** [comment]
@akoeplinger Ah, that explains it. I've updated the PR to throw the same exception as .NET on Desktop. Ideally, I'd like to run the across all incarnations of System.Drawing, so that would help.

**Pull request merged**